compiler/ilgen/VirtualMachineOperandStack.hpp

```diff
@@ -80,8 +80,31 @@ class VirtualMachineOperandStack : public VirtualMachineState
     * @param sizeHint initial size used to allocate the stack; will grow larger if needed
     * @param elementType TR::IlType representing the underlying type of the virtual machine's operand stack entries
     * @param stackTop previously allocated and initialized VirtualMachineRegister representing the top of stack
+    * Defaults behaviour for stack direction is "up" (push is sp++) and offset is -1 (push is *++sp = value)
+    * This is for compatibility with earlier versions of this constructor.
     */
    VirtualMachineOperandStack(TR::MethodBuilder *mb, int32_t sizeHint, TR::IlType *elementType, VirtualMachineRegister *stackTop);
+
+  /**
+    * @brief public constructor, must be instantiated inside a compilation because uses heap memory
+    * @param mb TR::MethodBuilder object of the method currently being compiled
+    * @param sizeHint initial size used to allocate the stack; will grow larger if needed
+    * @param elementType TR::IlType representing the underlying type of the virtual machine's operand stack entries
+    * @param stackTop previously allocated and initialized VirtualMachineRegister representing the top of stack
+    * @param growsUp to configure virtual machine stack growth direction, set to true if virtual machine stack grows towards larger addresses, false otherwise 
+    * @param stackInitialOffset to configure virtual machine stack stack offset 
+    * set to the difference in elements between initial stack pointer and actual bottom of stack
+    * Some stacks Push by incrementing stack pointer then storing, some by storing and then
+    * incrementing stack pointer. In the first case, stackPtrStartingOffset() should return -1
```

Comment:
please change the comments to correctly refer to the stackInitialOffset parameter's value rather than the old stackPtrStartingOffset() method's return value. Otherwise, looks good to me.